akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -759,6 +760,12 @@ private[akka] class ShardRegion(
         context.stop(self)
       }
 
+    case _: MemberReadyForShutdown | _: MemberPreparingForShutdown =>
+      if (!preparingForShutdown) {
+        log.info("{}. preparing for shutdown", typeName)
+      }
+      preparingForShutdown = true
+
```

Comment:
Shouldn't we send a `HandOff` to all shards under this region? 

Coordinator, Region and Shard are all listen to the events, setting their own `preparingForShutdown` and changing their behavior. But I don't see anything triggering the shutdown of entities. 

The shard do send a stop message to all active entities and then stop itself, but that only happens if someone sends it a `HandOff` message. 

We could let the region send a `GracefulShutdownReq` to coordinator and let it trigger the hand-offs without running a rebalance (cleanest as it follows the normal path). 

Alternatively, to speed up, we could let the region send the hand-off to its shards. Less network dance between the parties. 

Or perhaps the goal here is to just change the behavior and until for the cluster to be scaled to 0?

Reply:
We don't want to start any shutdown here, only know that when a shutdown happens to know we can shutdown quickly as the whole cluster is shutting down. The time between prepare to shutdown and the actual shutdown could be seconds.